src/NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp

```diff
@@ -25,40 +25,44 @@ template <size_t Dim>
 struct Mesh;
 template <class TagList>
 struct Variables;
-
-namespace Tags_detail {
-template <typename T, typename S, typename U,
-          typename = std::nullptr_t>
-struct deriv_impl;
-}  // namespace Tags_detail
 /// \endcond
 
 /*!
  * \ingroup DataBoxTagsGroup
- * \brief Prefix for a spatial derivative
+ * \brief Prefix indicating spatial derivatives
  *
- * There are three variants of this tag that change how the derivatives are
- * computed depending on which is chosen. The simplest is the non-compute
- * item version for a Tensor tag. This takes three template parameters:
- * 1. The tag to wrap
- * 2. The volume dim as a type (e.g. `tmpl::size_t<Dim>`)
- * 3. The frame of the derivative index
+ * Prefix indicating the spatial derivatives of a Tensor or that a Variables
+ * contains spatial derivatives of Tensors.
  *
- * The second variant of the derivative tag is a compute item that computes
- * the partial derivatives in a Frame that is not the logical frame. This
- * compute item is used by specifying the list of tags in the
- * `Tags::Variables<tmpl::list<Tags...>>`, the list of tags to be
- * differentiated, and the Tag for inverse Jacobian between the logical frame
- * and the frame in which the derivative is taken. Note that the derivative Tags
- * must be a contiguous subset from the head of the `Tags...` pack.
+ * \tparam Tag The tag to wrap
+ * \tparam Dim The volume dim as a type (e.g. `tmpl::size_t<Dim>`)
+ * \tparam Frame The frame of the derivative index
  *
- * The third variant of the derivative tag is a compute item that computes the
- * partial derivatives in the logical frame. In that case the template
- * parameters must be the list of Variables tags and derivative tags, with the
- * last parameter being a `std::integral_constant` of the dimension of the mesh.
+ * \see Tags::ComputeDeriv
  */
-template <typename T, typename S = void, typename U = void>
-struct deriv : Tags_detail::deriv_impl<T, S, U> {};
+template <typename Tag, typename Dim, typename Frame, typename = std::nullptr_t>
+struct deriv;
+
+template <typename Tag, typename Dim, typename Frame>
+struct deriv<Tag, Dim, Frame, Requires<tt::is_a_v<Tensor, db::item_type<Tag>>>>
+    : db::PrefixTag, db::SimpleTag {
+  using type =
+      TensorMetafunctions::prepend_spatial_index<db::item_type<Tag>, Dim::value,
+                                                 UpLo::Lo, Frame>;
+  using tag = Tag;
+  static std::string name() noexcept { return "deriv(" + Tag::name() + ")"; }
+  static constexpr bool should_be_sliced_to_boundary = true;
```

Comment:
The "should_be_sliced_to_boundary" variables were recently obsoleted (#921).  Remove them.